python/ray/data/dataset.py

```diff
@@ -2744,12 +2745,9 @@ def write_datasource(
         """
         if ray_remote_args is None:
             ray_remote_args = {}
+
         path = write_args.get("path", None)
-        if path and _is_local_scheme(path):
-            if ray.util.client.ray.is_connected():
-                raise ValueError(
-                    f"The local scheme paths {path} are not supported in Ray Client."
-                )
+        if path and not _has_scheme(path) or _is_local_scheme(path):
             ray_remote_args["scheduling_strategy"] = NodeAffinitySchedulingStrategy(
                 ray.get_runtime_context().get_node_id(),
                 soft=False,
```

Comment:
This code might make more sense in `FileBasedDatasource`, but you can't specify Ray remote args at the datasource abstraction layer.